library/.gitignore

```diff
@@ -6,3 +6,4 @@ libmbed*
 # Automatically generated files
 /error.c
 /version_features.c
+/state.c
```

Comment:
`state.c` is a very generic name. I would prefer a less vague name. Given the current content of this file, `ssl_states.c` would be more precise. This file could grow to have things other than one state-related function so `ssl_debug_helpers` would be reasonable as well. Since this file is in the SSL part of the library, it should have an `ssl_` prefix. I have a preference for something like `ssl_debug_helpers` because the file's purpose is more important than its contents. I've also been thinking we should have a naming convention for generated files, so something like `ssl_debug_helpers_generated.c`.

One practical reason to avoid non-specific names is that Mbed OS requires C modules to have unique names. In the past, we got away with `debug` and `error` but had to rename `mpi` (Mbed TLS's Multi Precision Integers collided with Mbed OS's Message Passing Interface). `debug` is a legacy name, and if it was new today we'd call it `ssl_debug`. `error` is also a legacy name and is supposed to be split into `{crypto,x509,ssl}_error` at some point in the future.

Reply:
Renamed to `ssl_debug_helpers_generated` and `generate_ssl_debug_helpers.py`